### ISSUE TITLE
Teams PR Review Checklist 

### ISSUE THREAD

**Issue by nemaniarjun** [AIcrowd/AIcrowd#987]
- [x] Missing New Team Button
- [x] New Teams Modal should check for constraints and warn before clicking on create
`Name may only contain letters and numbers and these quoted characters: "-_.{}[]"` 
- [x] Failure to create team should redirect to the challenge page not to root_path
- [x] Team names should allow spaces? what about apostrophes? I tried to create a team name "Arjun's MineRL Team" 
- [x] Challenge name on the profile teams tab should link to the challenge? 
- [x] How do I (Can I) Delete / Retire a team? We need this IMO!
- [x] Teams API Key, (we have it in the design docs, do we need this?)
- [x] @jsnrynlds can probably tell how to better display the submission warning on team page: https://aicrowd-production-pr-985.herokuapp.com/teams/arjun_learning_to_walk
- [x] Teams page does not show my profile image, but the stock image instead 

- [x] Tried to invite mohanty to my team, got a ugly 500 (not a pretty one)
Logs:
`
2019-07-26T20:33:08.803309+00:00 app[web.1]: 24 TID-gn4dyx2xs FATAL: NameError (uninitialized constant #<Class:0x00005566a9f788d8>::InvitationPendingNotifierJob
2019-07-26T20:33:08.803311+00:00 app[web.1]: Did you mean?  Team::InvitationDeclinedNotifierJob
2019-07-26T20:33:08.803312+00:00 app[web.1]: Team::InvitationCanceledNotifierJob):
`
- [x] We need to have basic tests if possible
- [x] Need to have Teams added to ActiveAdmin (Wont take more than 10 mins thanks to how awesome ActiveAdmin is)

Added some more comments on the PR

**Comment by nemaniarjun** [comment]
Suppose I make a submission as a single team, and then I join another team. What happens to my submission? Is it marked with my name or with the old team or new? 

Do we just aggregate submissions with teams instead of participants from now on while calculating the leaderboard?

**Comment by nemaniarjun** [comment]
Another weird error, not sure what is causing it. On the teams deploy, sign in as any user and then try logging out. It will give a 404 instead of logging the user out.

**Comment by nemaniarjun** [comment]
We should also add allow 'invite by email address'. If the invitee doesnt have an aicrowd account we send a different template inviting them to sign up and notify the inviter about the same. 

Also this allows people to quickly add people whose username they dont know.

**Comment by nemaniarjun** [comment]
Finally we had a stats icon for teams! We can show it now!!!! :smile:

**Comment by bkDJ** [comment]
As of 08e827ba12d95476b54aa8bf5abaa1f8dd1c6e29 Some of the above are fixed. I've marked them as such in the OP.
* Missing New Team Button
* Failure to create team should redirect to the challenge page not to root_path

**Comment by bkDJ** [comment]
@nemaniarjun to answer your above question about retiring a team, that in not planned for V1 of the team feature. For now, submissions are still tied to users, but the user being in a team ties the submission to the team for the leaderboard, even if the submission was made before the user joined a team. Users able to jump from one team to another but have their old submission tied to the old team would require some changes that are not planned as of yet. Also, preventing team organizers from disbanding the team (e.g. if their submission was the best to deprive the other team members of equal rewards) is useful.

In short: disallowing team deletion is in the specs and solves a host of tricky situations. We'll revisit the idea at a later date, surely.

**Comment by marcelsalathe** [comment]
lets no do the activeadmin for now, can do you later during an active admin revamp

**Comment by spMohanty** [comment]
Yes indeed. If all the major things are dealt with, we should move to the Leaderboard computation as soon as we can. That might take more time that we expect, so good to get started on that early.

**Comment by bkDJ** [comment]
All of the points addressed as of 05f59f015d5499f2761f1822a39232d172bb6e50
